packages/plugin-ext/src/plugin/authentication-ext.ts

```diff
@@ -0,0 +1,219 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+// code copied and modified from https://github.com/microsoft/vscode/blob/1.47.3/src/vs/workbench/api/common/extHostAuthentication.ts
+
+import { Disposable } from './types-impl';
+import {
+    AuthenticationExt,
+    AuthenticationMain, Plugin as InternalPlugin,
+    PLUGIN_RPC_CONTEXT
+} from '../common/plugin-api-rpc';
+import { RPCProtocol } from '../common/rpc-protocol';
+import { Emitter, Event } from '@theia/core/lib/common/event';
+import * as theia from '@theia/plugin';
+import { AuthenticationSessionsChangeEvent } from '@theia/core/lib/browser/authentication-service';
+
+export class AuthenticationExtImpl implements AuthenticationExt {
+    private proxy: AuthenticationMain;
+    private authenticationProviders: Map<string, theia.AuthenticationProvider> = new Map<string, theia.AuthenticationProvider>();
+
+    private _providerIds: string[] = [];
+
+    private _providers: theia.AuthenticationProviderInformation[] = [];
+
+    private onDidChangeAuthenticationProvidersEmitter = new Emitter<theia.AuthenticationProvidersChangeEvent>();
+    readonly onDidChangeAuthenticationProviders: Event<theia.AuthenticationProvidersChangeEvent> = this.onDidChangeAuthenticationProvidersEmitter.event;
+
+    private onDidChangeSessionsEmitter = new Emitter<theia.AuthenticationSessionsChangeEvent>();
+    readonly onDidChangeSessions: Event<theia.AuthenticationSessionsChangeEvent> = this.onDidChangeSessionsEmitter.event;
+
+    constructor(rpc: RPCProtocol) {
+        this.proxy = rpc.getProxy(PLUGIN_RPC_CONTEXT.AUTHENTICATION_MAIN);
+    }
+
+    getProviderIds(): Promise<ReadonlyArray<string>> {
+        return this.proxy.$getProviderIds();
+    }
+
+    get providerIds(): string[] {
+        return this._providerIds;
+    }
+
+    get providers(): ReadonlyArray<theia.AuthenticationProviderInformation> {
+        return Object.freeze(this._providers.slice());
+    }
+
+    async getSession(requestingExtension: InternalPlugin, providerId: string, scopes: string[],
+                     options: theia.AuthenticationGetSessionOptions & { createIfNone: true }): Promise<theia.AuthenticationSession>;
+    async getSession(requestingExtension: InternalPlugin, providerId: string, scopes: string[],
+                     options: theia.AuthenticationGetSessionOptions = {}): Promise<theia.AuthenticationSession | undefined> {
+        const provider = this.authenticationProviders.get(providerId);
+        const extensionName = requestingExtension.model.displayName || requestingExtension.model.name;
+        const extensionId = requestingExtension.model.id.toLowerCase();
+
+        if (!provider) {
+            return this.proxy.$getSession(providerId, scopes, extensionId, extensionName, options);
```

Comment:
done

Reply:
This will not work if the authentication providers are not in the same plugin host: what I meant was to always delegate to the main side and remove the handling here.

Reply:
Fixed, used the logic from main